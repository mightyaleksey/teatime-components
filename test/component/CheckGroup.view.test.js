'use strict';

const CheckGroup = require('../../component/CheckGroup');
const React = require('react');
const renderer = require('react-test-renderer');

test('defaultProp #size should have a value `m`', () => {
  expect(CheckGroup.defaultProps.size).toBe('m');
});

test('Renders correctly with default props', () => {
  const tree = renderer.create(
    <CheckGroup
      name='control'
      options={[
        {label: 'AA', value: 'aa'},
        {label: 'BB', value: 'bb'},
        {label: 'CC', value: 'cc'},
      ]}/>
  );

  expect(tree).toMatchSnapshot();
});

test('Renders correctly with defined props', () => {
  const tree = renderer.create(
    <CheckGroup
      className='outer'
      cols={2}
      disabled
      id='controls-id'
      name='control'
      options={[
        {label: 'AA', value: 'aa'},
        {label: 'BB', value: 'bb'},
        {label: 'CC', value: 'cc'},
      ]}
      value={[
        'bb',
        'cc',
      ]}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
