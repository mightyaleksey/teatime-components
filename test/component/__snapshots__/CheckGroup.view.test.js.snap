// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with default props 1`] = `
<div
  className="container"
  id={undefined}
  onChange={undefined}
>
  <div
    className="column"
  >
    <span
      className="wrapper"
    >
      <input
        checked={false}
        className="native"
        defaultValue={undefined}
        disabled={undefined}
        id="_teatime1"
        name="control"
        onChange={[Function]}
        type="checkbox"
        value="aa"
      />
      <label
        className="control"
        htmlFor="_teatime1"
      />
      <label
        className="label"
        htmlFor="_teatime1"
      >
        AA
      </label>
    </span>
    <span
      className="wrapper"
    >
      <input
        checked={false}
        className="native"
        defaultValue={undefined}
        disabled={undefined}
        id="_teatime2"
        name="control"
        onChange={[Function]}
        type="checkbox"
        value="bb"
      />
      <label
        className="control"
        htmlFor="_teatime2"
      />
      <label
        className="label"
        htmlFor="_teatime2"
      >
        BB
      </label>
    </span>
    <span
      className="wrapper"
    >
      <input
        checked={false}
        className="native"
        defaultValue={undefined}
        disabled={undefined}
        id="_teatime3"
        name="control"
        onChange={[Function]}
        type="checkbox"
        value="cc"
      />
      <label
        className="control"
        htmlFor="_teatime3"
      />
      <label
        className="label"
        htmlFor="_teatime3"
      >
        CC
      </label>
    </span>
  </div>
</div>
`;

exports[`Renders correctly with defined props 1`] = `
<div
  className="container outer"
  id="controls-id"
  onChange={undefined}
>
  <div
    className="column"
  >
    <span
      className="wrapper"
    >
      <input
        checked={false}
        className="native"
        defaultValue={undefined}
        disabled={true}
        id="_teatime4"
        name="control"
        onChange={[Function]}
        type="checkbox"
        value="aa"
      />
      <label
        className="control"
        htmlFor="_teatime4"
      />
      <label
        className="label"
        htmlFor="_teatime4"
      >
        AA
      </label>
    </span>
    <span
      className="wrapper"
    >
      <input
        checked={true}
        className="native"
        defaultValue={undefined}
        disabled={true}
        id="_teatime5"
        name="control"
        onChange={[Function]}
        type="checkbox"
        value="bb"
      />
      <label
        className="control"
        htmlFor="_teatime5"
      />
      <label
        className="label"
        htmlFor="_teatime5"
      >
        BB
      </label>
    </span>
  </div>
  <div
    className="column"
  >
    <span
      className="wrapper"
    >
      <input
        checked={true}
        className="native"
        defaultValue={undefined}
        disabled={true}
        id="_teatime6"
        name="control"
        onChange={[Function]}
        type="checkbox"
        value="cc"
      />
      <label
        className="control"
        htmlFor="_teatime6"
      />
      <label
        className="label"
        htmlFor="_teatime6"
      >
        CC
      </label>
    </span>
  </div>
</div>
`;
